adagrams/game.py

```diff
@@ -1,11 +1,95 @@
+import random
+
 def draw_letters():
-    pass
+    LETTER_POOL = {
+    'A': 9, 'B': 2, 'C': 2, 'D': 4, 'E': 12, 'F': 2, 'G': 3, 'H': 2, 
+    'I': 9, 'J': 1, 'K': 1, 'L': 4, 'M': 2, 'N': 6, 'O': 8, 'P': 2, 
+    'Q': 1, 'R': 6, 'S': 4, 'T': 6, 'U': 4, 'V': 2, 'W': 2, 'X': 1, 
+    'Y': 2, 'Z': 1
+    }
+    MAX_HAND_COUNT = 10
+    all_available_let = []
+    current_hand = []
+
+    # creates a list of avaialble 98 letters based on respective frequencis
+    for letter, original_letter_count in LETTER_POOL.items():
+        for letter_index in range(original_letter_count):
+            all_available_let.append(letter)
+
+    # updates initial available list of letters to show remaining letters
+    while len(current_hand) < MAX_HAND_COUNT:
+        available_let_list_index = random.randint(0, len(all_available_let)-1)
+        current_hand.append(all_available_let[available_let_list_index])
+        all_available_let.pop(available_let_list_index)
+
+    return current_hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    word = word.upper() # capitalizes input word
+
+    # assures that each letter in the word is in the current word bank and within the count
+    for input_letter in word:
+        letter_freq_in_letter_bank = letter_bank.count(input_letter)
+        letter_freq_in_word = word.count(input_letter)
```

Comment:
Under the hood, `.count()` is essentially looping over the iterable that provided it and counts the instances of its provided elements (`input_letter`). Essentially you are looping over `word` twice but this same functionality could be provided without looping over `word` with count. How do you think this could be done?